Source/Signal.swift

```diff
@@ -109,3 +109,73 @@ extension SignalType where T: SequenceType {
         }
     }
 }
+
+// MARK: Boolean Operations
+
+extension SignalType where T: BooleanType {    
+    /// Returns a signal that inverts each value sent on self
+    @warn_unused_result(message="Did you forget to call `observe` on the signal?")
+    public func not() -> Signal<Bool, E> {
+        return Signal { observer in
+            return self.observe { (event: Event<T, E>) in
+                observer(event.map { !$0 })
+            }
+        }
+    }
+}
+
+extension SignalType where T: SequenceType, T.Generator.Element: BooleanType {
+    /// Returns a signal that sends `a && b && c && ...` for each value (a, b, c, ...) sent on self
+    @warn_unused_result(message="Did you forget to call `observe` on the signal?")
+    public func and() -> Signal<Bool, E> {
+        return Signal<Bool, E> { observer in
+            return self.observe { event in
+                switch event {
+                case let .Next(sequence):
+                    sendNext(observer, sequence.reduce(true) { $0.boolValue && $1.boolValue })
+                case let .Error(error):
+                    sendError(observer, error)
+                case .Completed:
+                    sendCompleted(observer)
+                case .Interrupted:
+                    sendInterrupted(observer)
+                }
+            }
+        }
+    }
+
+    /// Returns a signal that sends `a ||& b || c || ...` for each value (a, b, c, ...) sent on self
+    @warn_unused_result(message="Did you forget to call `observe` on the signal?")
+    public func or() -> Signal<Bool, E> {
+        return Signal<Bool, E> { observer in
+            return self.observe { event in
+                switch event {
+                case let .Next(sequence):
+                    sendNext(observer, sequence.reduce(false) { $0.boolValue || $1.boolValue })
+                case let .Error(error):
+                    sendError(observer, error)
+                case .Completed:
+                    sendCompleted(observer)
+                case .Interrupted:
+                    sendInterrupted(observer)
+                }
+            }
+        }
+    }
+}
+
+extension SignalType where T == (Bool, Bool) {
```

Comment:
I tried to implement this as `extension SignalType where T == (B1: BooleanType, B2: BooleanType)`, but then Xcode gives an error: 
> Ambiguous reference to member `and`

at the callsite, with 2 times "Found this candidate" details, but for both entries, Xcode just displays the callsite instead of the candidate :disappointed: 

So I'm not sure if I'm doing something wrong here, or if thats broken in Xcode...